src/Draco.Compiler/Api/Semantics/Symbol.cs

```diff
@@ -178,9 +210,38 @@ public ModuleSymbol(Internal.Symbols.ModuleSymbol module)
     }
 }
 
+internal sealed class FieldSymbol : SymbolBase<Internal.Symbols.FieldSymbol>, IFieldSymbol
+{
+    public bool IsMutable => this.Symbol.IsMutable;
+    public bool IsStatic => this.Symbol.IsStatic;
+    public ITypeSymbol Type => (ITypeSymbol)this.Symbol.Type.ToApiSymbol();
```

Comment:
You can get rid of casting, just make the override in `FieldSymbol` return an `IFieldSymbol`. If you see any casts like this, feel free to use covariant overrides. It's just that there was a time when the compiler targeted ns2.0 and there was more limited support.